rivertype/job_row.go

```diff
@@ -108,9 +110,21 @@ const (
 	JobStateScheduled JobState = "scheduled"
 )
 
+// AttemptError is an error from a single job attempt that failed due to an
+// error or a panic.
 type AttemptError struct {
-	At    time.Time `json:"at"`
-	Error string    `json:"error"`
-	Num   int       `json:"num"`
-	Trace string    `json:"trace"`
+	// At is the time at which the error occurred.
+	At time.Time `json:"at"`
+
+	// Attempt is the attempt number on which the error occurred (maps to
+	// Attempt on a job row).
+	Attempt int `json:"attempt"`
```

Comment:
@bgentry Wanted to float this rename of `Num` to `Attempt`. I had to look up what was being set to `Num` and realized that it'd be more obvious if this had the same name as `JobRow.Attempt`, which is what produces this number. I feel like it's just one of these things that we had since the beginning and never reevaluated. What do you think?

Reply:
I think it’s a nice thinking improvement at the small cost of a few more bytes for each error in the db. Given we’re serializing stack traces that’s an insignificant difference.

Related, should we be limiting the buffer size for stack traces somehow? These could result in very large payloads.

Reply:
+1.

And yeah ... we should take a look at just what the average size of one of those panic traces looks like again. I guess the good news is that errors should be the rarer case, panics should be the rarer case within errors, and long strings like this all get shoved in TOAST anyway, so it shouldn't affect the actual table itself every much.

But there might be something relatively smart/easy, low-hanging that we can do to improve things. I'll add a TODO note.